File: MAStaff-Bukkit/src/main/java/es/angelillo15/mast/bukkit/cmd/mast/ReloadARG.java
package es.angelillo15.mast.bukkit.cmd.mast;

import com.nookure.mast.api.cmd.SubCommand;
import com.nookure.mast.api.cmd.sender.CommandSender;
import es.angelillo15.mast.api.config.bukkit.Messages;
import es.angelillo15.mast.bukkit.MAStaff;
import org.jetbrains.annotations.NotNull;

public class ReloadARG extends SubCommand {
  @Override
  public String getName() {
    return "reload";
  }

  @Override
  public String getDescription() {
    return "Reloads the plugin";
  }

  @Override
  public String getSyntax() {
    return "/mast reload";
  }

  @Override
  public String getPermission() {
    return "mast.reload";
  }

  @Override
  public void onCommand(@NotNull CommandSender sender, @NotNull String label, @NotNull String[] args) {
    MAStaff.getPlugin().reload();
    sender.sendMessage(Messages.GET_RELOADED_MESSAGE());
  }
}

File: MAStaff-Velocity/src/main/java/es/angelillo15/mast/velocity/cmd/mastv/ReloadArg.kt
package es.angelillo15.mast.velocity.cmd.mastv

import com.google.inject.Inject
import es.angelillo15.mast.api.cmd.SubCommand
import es.angelillo15.mast.api.cmd.sender.CommandSender
import es.angelillo15.mast.api.config.velocity.Messages
import es.angelillo15.mast.api.utils.MAStaffInject

class ReloadArg : SubCommand() {
    @Inject
    private lateinit var plugin: MAStaffInject
    override val name: String
        get() = "reload"
    override val description: String
        get() = "Reloads the plugin"
    override val syntax: String
        get() = "/mastv reload"
    override val permission: String
        get() = "mast.reload"

    override fun onCommand(sender: CommandSender, label: String, args: Array<String>) {
        plugin.reload()
        sender!!.sendMessage(Messages.reloaded());
    }
}